src/herder/test/UpgradesTests.cpp

```diff
@@ -2363,6 +2627,46 @@ TEST_CASE("upgrade from cpp14 serialized data", "[upgrades]")
     REQUIRE(!up.mBaseReserve.has_value());
 }
 
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+TEST_CASE("upgrades serialization roundtrip", "[upgrades]")
```

Comment:
I removed that section because it was failing due to a null `assumeStateWork` in `LedgerManagerImpl` on startup. I **think** it was also failing in your PR with the same error (but I can't remember now so let me confirm). I added it on a list of tests to add back in.

Reply:
Yeah that test segfaults in your PR as well. I'm assuming it used to work for you?

Reply:
Yeah, it definitely used to work for me; there is no way I would have manually come up with the expected JSON output. It's weird it segfaults now; probably something has been changed a while ago...

Reply:
I added the debug string and added a test, but the test doesn't "restart" app to test persisted upgrades.